### PR TITLE
Fix issue on building techinfo

### PR DESCRIPTION
**Pull request by Chewbaka69**
When the ressource factor is reduce or put to 0, the techinfo table of the building production is devide by the percentage

image

### CODE REVIEW

File: app/Services/PlanetService.php [hunk 1]
```diff
@@ -1163,7 +1163,7 @@ private function updateResourceProductionStatsInner(Resources $production_total,
      * @return Resources
      * @throws Exception
      */
-    public function getBuildingProduction(string $machine_name, int $building_level = 0): Resources
+    public function getBuildingProduction(string $machine_name, int|null $building_level = null, bool $force_factor = false): Resources
     {
```
Comment (lanedirt) [inline app/Services/PlanetService.php:1170]: Could you please update the phpdoc comment to include the new parameter.

**Review by lanedirt — CHANGES_REQUESTED** [review]

**Comment by lanedirt** [comment]
Nice find, thanks for the fix. I have one small remark regarding missing phpdoc argument, other than that looks good!

File: app/Services/PlanetService.php (hunk 1 above)
Comment (Chewbaka69) [reply app/Services/PlanetService.php:1170]: Oh yes sure my bad

Comment (lanedirt) [reply app/Services/PlanetService.php:1170]: Super, thx!

**Review by lanedirt — APPROVED** [review]

**Pull request merged**